MRC23.md

```diff
@@ -0,0 +1,344 @@
+
+
+### Smart Agent SDK Architecture [WIP]
+
+The [Smart Agent Protocol White Paper](https://github.com/SmartAgentProtocol/SmartAgents/blob/main/White%20Paper.md) provides a high-level overview of the function and incentive mechanism for Smart Agents, and the [Smart Agent repo](https://github.com/MorpheusAIs/SmartAgents) contains an initial agent implementation. We extend this work and propose the development of a Smart Agent SDK that will create a flexible foundation for agent developers to build upon, addressing a number of common needs without prescribing any specific agent implementation, framework, or models.  This may be delivered as a v2 of the existing Smart Agent project.
+
+
+
+![alt_text](https://mor-assets.s3.amazonaws.com/smartagentarch.png "image_tooltip")
+
+
+The diagram shows the key components that the Smart Agent SDK is concerned with. It is initially expected to run entirely locally but may be **progressively decentralized from the bottom-up.**  I.e. first storage and compute, then models (inference) and indexes as these services become available.
+
+A Smart Agent may run entirely stand-alone, or it may register itself with the Morpheus network, in which case it can coordinate with other agents and offer its services to the network.  
+
+It should be **agent orchestration framework agnostic**, but provide adapters for common frameworks such as langchain, AutoGPT, Crew AI so that agent developers can use their favorite framework.  The SDK will also provide a light wrapper for tools, making them easily reusable by other Smart Agent developers.
+
+
+#### Registration and discovery
+
+Agents should provide a manifest that contains all the details a user needs in order to assess whether it is suitable for their needs.
+
+Best practices should be developed for disclosing any data sources or potential quality issues as these propagate downstream. This disclosure process also enables auditability - independent auditors can assess whether the agent does what it says it does.
+
+Agents may register themselves with a central smart contract (see proposed work by [Lumerin](https://lumerin.io)).  There should be a decentralized reputation system overlaid on this to help users decide which agents to trust.  This smart contract will act as the registry for an agent marketplace.
+
+See: [MRC 10: Development of Smart Agent Tools Marketplace](https://docs.google.com/document/d/1JJQR4-b63VbJQ_etSt5Poyk1KcK3f7FvUZcwKB5Zpk8/edit).
+
+
+#### Tool Use (and re-use)
+
+The SDK should aim to encourage tool code reuse by enabling compatibility of tools between competing agent frameworks, and providing an open source repo to share tools.
+
+The interface for a tool can be very light, including at at minimum:
+
+
+
+* Purpose of the tool
+* Input format
+* Output format
```

Comment:
Is it worth adding a bullet for `* n examples`, where we explicitly ask for several examples for every agent, because these often greatly improve the prompt response of LLM Agents. 

For example:
```
* three examples:
   * _example input 1_ -> _example output 1_
   * ...
```

Reply:
Good point, I think there are probably a number of things that will end up in here but I'd rather not try to be exhaustive here as the goal is to keep it high level and not get into the weeds with implementation details.